pkg/api/usecase/wishboard/interactor.go

```diff
@@ -0,0 +1,228 @@
+	wishBoardService wishboardservice.Service
+	fileService      fileservice.Service
+}
+
+func New(masterTxManager repository.MasterTxManager, userService userservice.Service, wishBoardService wishboardservice.Service, fileService fileservice.Service) Interactor {
+	return &interactor{
+		masterTxManager:  masterTxManager,
+		userService:      userService,
+		wishBoardService: wishBoardService,
+		fileService:      fileService,
+	}
+}
+
+func (i *interactor) CreateNewWishBoard(ctx context.Context, authID, title string, backgroundImage []byte) (*wishboard.Entity, error) {
+	if err := validateTitle(ctx, title); err != nil {
+		return nil, werrors.Stack(err)
+	}
+
+	var wishBoardEntity *wishboard.Entity
+	err := i.masterTxManager.Transaction(ctx, func(ctx context.Context, masterTx repository.MasterTx) error {
+		u, err := i.userService.GetByAuthID(ctx, masterTx, authID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		// 背景画像を保存し、URLを取得
+		backgroundImageURL, err := i.fileService.UploadImageToLocalFolder(backgroundImage)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		// TODO: 招待URLの自動生成
+		inviteURL := "hoge"
+
+		wishBoardEntity, err = i.wishBoardService.Create(ctx, masterTx, title, backgroundImageURL, inviteURL, u.ID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
+
+	return wishBoardEntity, nil
+}
+
+func (i *interactor) GetMyWishBoards(ctx context.Context, authID string) (wishboard.EntitySlice, error) {
+	var wishBoardEntitySlice wishboard.EntitySlice
+	err := i.masterTxManager.Transaction(ctx, func(ctx context.Context, masterTx repository.MasterTx) error {
+		userEntity, err := i.userService.GetByAuthID(ctx, masterTx, authID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		// 自分が所属しているWishBoardのリストを取得
+		wishBoardEntitySlice, err = i.wishBoardService.GetMyBoards(ctx, masterTx, userEntity.ID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
+
+	return wishBoardEntitySlice, nil
+}
+
+func (i *interactor) UpdateTitle(ctx context.Context, wishBoardID int, title, authID string) error {
+	if err := validateTitle(ctx, title); err != nil {
+		return werrors.Stack(err)
+	}
+
+	err := i.masterTxManager.Transaction(ctx, func(ctx context.Context, masterTx repository.MasterTx) error {
+		userEntity, err := i.userService.GetByAuthID(ctx, masterTx, authID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		// WishBoardが存在するか確認
+		wishBoardEntity, err := i.wishBoardService.GetByPK(ctx, masterTx, wishBoardID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+
+		// ユーザがWishBoardのメンバーでなければPermissionDenied
+		isMember, err := i.wishBoardService.IsMember(ctx, masterTx, userEntity.ID, wishBoardEntity.ID)
+		if err != nil {
+			return werrors.Stack(err)
+		}
+		if !isMember {
+			err := errors.New("you don't belong to wish_board")
```

Comment:
`errors.New`はあくまでログ出力用のエラーであって、ユーザーに返すエラーではないから、 `「permission denied」`でも良さそうやね！

```suggestion
			err := errors.New("Error occurred when update board title. cause: permission denied")
```